boskos/metrics/metrics.go

```diff
@@ -0,0 +1,171 @@
+				Name: "boskos_gke_project_free",
+				Help: "Number of free gke-project",
+			}),
+			"busy": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_busy",
+				Help: "Number of busy gke-project",
+			}),
+			"dirty": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_dirty",
+				Help: "Number of dirty gke-project",
+			}),
+			"cleaning": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_cleaning",
+				Help: "Number of cleaning gke-project",
+			}),
+		},
+	}
+)
+
+func init() {
+	for _, gce := range promMetrics.GceStats {
+		prometheus.MustRegister(gce)
+	}
+
+	for _, gke := range promMetrics.GkeStats {
+		prometheus.MustRegister(gke)
+	}
+}
+
+func main() {
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	boskos := client.NewClient("Reaper", "http://boskos")
+	logrus.Infof("Initialzied boskos client!")
+
+	http.Handle("/prometheus", promhttp.Handler())
+	http.Handle("/", handleMetric(boskos))
+
+	go func() {
+		logTick := time.NewTicker(time.Minute).C
+		for {
+			select {
+			case <-logTick:
+				if err := update(boskos); err != nil {
+					logrus.WithError(err).Warning("[Boskos Metrics]Update failed!")
+				}
+			}
+		}
+	}()
+
+	logrus.Info("Start Service")
+	logrus.WithError(http.ListenAndServe(":8080", nil)).Fatal("ListenAndServe returned.")
+}
+
+func update(boskos *client.Client) error {
+	if gce, err := boskos.Metric("gce-project"); err != nil {
+		return fmt.Errorf("fail to get metric for gce-project : %v", err)
+	} else {
+		promMetrics.GceStats["free"].Set(float64(gce.Current["free"]))
+		promMetrics.GceStats["busy"].Set(float64(gce.Current["busy"]))
+		promMetrics.GceStats["dirty"].Set(float64(gce.Current["dirty"]))
+		promMetrics.GceStats["cleaning"].Set(float64(gce.Current["cleaning"]))
+	}
+
+	if gke, err := boskos.Metric("gke-project"); err != nil {
+		return fmt.Errorf("fail to get metric for gke-project : %v", err)
+	} else {
+		promMetrics.GkeStats["free"].Set(float64(gke.Current["free"]))
+		promMetrics.GkeStats["busy"].Set(float64(gke.Current["busy"]))
+		promMetrics.GkeStats["dirty"].Set(float64(gke.Current["dirty"]))
+		promMetrics.GkeStats["cleaning"].Set(float64(gke.Current["cleaning"]))
+	}
+
+	return nil
+}
+
+//  handleMetric: Handler for /metric
+//  Method: GET
+func handleMetric(boskos *client.Client) http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		logrus.WithField("handler", "handleMetric").Infof("From %v", req.RemoteAddr)
+
+		if req.Method != "GET" {
+			logrus.Warning("[BadRequest]method %v, expect GET", req.Method)
+			http.Error(res, "only accepts GET request", http.StatusMethodNotAllowed)
+			return
+		}
+
+		rtype := req.URL.Query().Get("type")
+		if rtype == "" {
+			msg := "type must be set in the request."
+			logrus.Warning(msg)
+			http.Error(res, msg, http.StatusBadRequest)
+			return
+		}
+
+		logrus.Infof("Request for metric %v", rtype)
+
+		metric, err := boskos.Metric(rtype)
```

Comment:
You could just forward the body of the http request to the boskos deployment and then return the body of the response. You don't need to unmarshal and then remarshal the `common.Metric` struct. Unmarshalling the data does let you log info about the response you are providing, but that could be logged from the boskos deployment as well.